pkg/flow/temporary.go

```diff
@@ -400,6 +432,8 @@ func (engine *engine) newIsolateRequest(ctx context.Context, im *instanceMemory,
 
 	ar := new(functionRequest)
 	ar.ActionID = uid.String()
+	ar.Originator = im.orginator()
+	ar.Iterator = im.iterator() // TODO
```

Comment:
TODO what?

Reply:
if we have a foreach loop nested in a foreach loop then the `iterator` value will be overwritten